### PR TITLE
Added template for Huawei SmartAX: display_ont_port_vlan_0_1_byport_eth_0

### PR DESCRIPTION
**Pull request by evilmonkey19**
The idea with the extra empty empty value is to capture the native_vlan and other values that cannot be capture. I am open to any suggestion on how to improve.

### CODE REVIEW

**Comment by jmcgill298** [comment]
> The idea with the extra empty empty value is to capture the native_vlan and other values that cannot be capture. I am open to any suggestion on how to improve.

@evilmonkey19 what do you mean by extra empty value, and by values that cannot be captured?

**Comment by evilmonkey19** [comment]
If you look at the .yml file you will notice that the "native_vlan" value is fulfilled only once. I do this because of the output. I get the values from the list and the extra values (native_vlan and so on) are filled in an another value.

**Comment by jmcgill298** [comment]
> If you look at the .yml file you will notice that the "native_vlan" value is fulfilled only once. I do this because of the output. I get the values from the list and the extra values (native_vlan and so on) are filled in an another value.

Oh, I think you are looking for Fillup:

```Value Fillup NATIVE_VLAN (...)```

**Comment by evilmonkey19** [comment]
I have tried before but unfortunately i didn't make it work and I had no clue on how to make it work :'( 
This is the template:
```
Value PORT_ID (\d+)
Value PORT_TYPE (\S+)
Value C_VLAN (\d+)
Value C_PRI (\d+)
Value ETH_TYPE (\S+)
Value VLAN_TYPE (\S+)
Value S_VLAN (\d+)
Value S_PRI (\d+)
Value S_PRI_POLICY (\S+)
Value Fillup NATIVE_VLAN (\d+)
Value Fillup DEFAULT_PRIORITY (\d+)
Value Fillup DOWNSTREAM_MODE (\S+)
Value Fillup MISMATCH_POLICY (\S+)

Start
  ^\s*Port\s*Port\s*C-VLAN\s*C-PRI\s*ETH-type\s*VLAN-type\s*S-VLAN\s*S-PRI\s*S-PRI\s*
  ^\s*type\s*ID\s*POLICY
  ^\s+${PORT_TYPE}\s+${PORT_ID}\s+(-|${C_VLAN})\s+(-|${C_PRI})\s+(-|${ETH_TYPE})\s+(-|${VLAN_TYPE})\s+(-|${S_VLAN})\s*(-|${S_PRI})\s*(-|${S_PRI_POLICY})\s* -> Record
  ^\s*Native\s*VLAN\s*:\s*${NATIVE_VLAN}\s*
  ^\s*Default\s*priority\s*:\s*${DEFAULT_PRIORITY}\s*
  ^\s*Downstream\s*mode\s*:\s*${DOWNSTREAM_MODE}\s*
  ^\s*Mismatch\s*policy\s*:\s*${MISMATCH_POLICY}\s*
  ^\s*
  ^. -> Error
```

**Comment by jmcgill298** [comment]
I sent a PR https://github.com/evilmonkey19/ntc-templates/pull/1 that passes tests for me

**Comment by evilmonkey19** [comment]
I see! It works as expected but again it adds 3 values to the list instead of only 2, although the list is only 2 rows. To me, it looks fine this case also 😃. I gotta say that Huawei made this case hard xD

**Comment by jmcgill298** [comment]
Oh, the additional entry can also be fixed

File: tests/huawei_smartax/display_ont_port_vlan_0_1_byport_eth_0/huawei_smartax_display_ont_port_vlan_0_1_byport_eth_0.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    downstream_mode: "operation"
+    eth_type: ""
+    mismatch_policy: "discard"
+    native_vlan: "1"
+    port_id: ""
+    port_type: ""
+    s_pri: ""
+    s_pri_policy: ""
+    s_vlan: ""
+    vlan_type: ""
```
Comment (jmcgill298) [inline tests/huawei_smartax/display_ont_port_vlan_0_1_byport_eth_0/huawei_smartax_display_ont_port_vlan_0_1_byport_eth_0.yml:41]:
```suggestion
```

File: ntc_templates/templates/huawei_smartax_display_ont_port_vlan_0_1_byport_eth_0.textfsm [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+Start
+  ^\s*Port\s*Port\s*C-VLAN\s*C-PRI\s*ETH-type\s*VLAN-type\s*S-VLAN\s*S-PRI\s*S-PRI\s*
+  ^\s*type\s*ID\s*POLICY
+  ^\s+${PORT_TYPE}\s+${PORT_ID}\s+(-|${C_VLAN})\s+(-|${C_PRI})\s+(-|${ETH_TYPE})\s+(-|${VLAN_TYPE})\s+(-|${S_VLAN})\s*(-|${S_PRI})\s*(-|${S_PRI_POLICY})\s* -> Record
+  ^\s*Native\s*VLAN\s*:\s*${NATIVE_VLAN}\s*
+  ^\s*Default\s*priority\s*:\s*${DEFAULT_PRIORITY}\s*
+  ^\s*Downstream\s*mode\s*:\s*${DOWNSTREAM_MODE}\s*
+  ^\s*Mismatch\s*policy\s*:\s*${MISMATCH_POLICY}\s* 
+  ^\s*
+  ^. -> Error
```
Comment (jmcgill298) [inline ntc_templates/templates/huawei_smartax_display_ont_port_vlan_0_1_byport_eth_0.textfsm:24]:
```suggestion
  ^. -> Error

EOF
```

**Review by jmcgill298 — APPROVED** [review]

**Comment by jmcgill298** [comment]
@evilmonkey19 there are two ways to solve that, either overriding the default behavior of `EOF -> Record` with just `EOF` to exclude the Record, or making one of the the capture groups `Required` so that the EOF->Record doesn't have a qualified entry to record

**Comment by evilmonkey19** [comment]
Awesome! Thanks for letting me know ❤️  It was the first time for me with this kind of templates

**Pull request merged**